src-java/reroute-topology/reroute-storm-topology/src/main/java/org/openkilda/wfm/topology/reroute/service/RerouteService.java

```diff
@@ -359,6 +359,27 @@ public void processManualRerouteRequest(MessageSender sender, String correlation
         sender.emitManualRerouteCommand(request.getFlowId(), flowThrottlingData);
     }
 
+    /**
+     * Handles request to update single switch flow status.
+     */
+    public void processSingleSwitchFlowStatusUpdate(UpdateSingleSwitchFlows request) {
+        Collection<Flow> affectedFlows = flowRepository.findOneSwitchFlows(request.getSwitchId());
+        FlowStatus newFlowStatus = request.getStatus() == SwitchStatus.ACTIVE ? FlowStatus.UP : FlowStatus.DOWN;
+        FlowPathStatus newFlowPathStatus = request.getStatus() == SwitchStatus.ACTIVE
+                ? FlowPathStatus.ACTIVE : FlowPathStatus.INACTIVE;
+        for (Flow flow : affectedFlows) {
+            log.info("Updating flow and path statuses for flow {} to {}, {}", flow.getFlowId(), newFlowStatus,
+                    newFlowPathStatus);
+            transactionManager.doInTransaction(() -> {
```

Comment:
Why not a transaction over all flows not over each of them?

Reply:
Since we need to have consistent state within flow/flow_path, not all flow/flow_paths on the switch, and it also reduce time for each transaction